staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -891,6 +913,25 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	return ret, nil
 }
 
+func defaultDeprecationWarning(deprecatedVersion string, crd apiextensionsv1.CustomResourceDefinitionSpec) string {
```

Comment:
versions for CRDs mapping to kube versions somehow seems weird.  I'd rather avoid this entire method by requiring the deprecation string.

Reply:
This is following the same rules we already follow when determining the order we present the CRD versions in discovery